### ISSUE TITLE
Clarification of the xcomp definition

### ISSUE THREAD

**Issue by makazhan** [UniversalDependencies/docs#200]
The definition of xcomp reads:
_An open clausal complement (xcomp) of a verb or an adjective is a predicative or clausal complement without its own subject. The reference of the subject is necessarily determined by an argument external to the xcomp (normally by the object of the next higher clause, if there is one, or else by the subject of the next higher clause)._
And there’s also an additional remark regarding covert subjects:
_Note that the above condition “without its own subject” does not mean that a clause is an xcomp just because its subject is not overt._
But what if a complement clause does not have a subject (overt or otherwise), but there’s a possibility to embed one.
For instance, in the following Kazakh sentence `кино көргенді` (watching movies) is a complement clause without a subject. Its subject can only be interpreted as `мен` (I):

```
Мен  кино    көргенді    ұнатамын
 I  movie  watching-ACC   like
```

But we still can embed a genitive subject putting the participle `көрген` in an appropriate possessive form:

```
Мен  оның   кино     көргенін        ұнатамын
 I  he.GEN  movie  watching-P3Sg-ACC   like
Мен  сенің   кино     көргеніңді      ұнатамын
 I  you.GEN  movie  watching-P2Sg-ACC   like
```

So the question is, in the original sentence, should we use `xcomp(көргенді, ұнатамын)` (per definition  : external subject with no possible _interpretation_) or `ccomp(көргенді, ұнатамын)` (per _possibility_ of having own/internal subject)?

In other words, if the subject of a complement clause cannot be interpreted as anything else than a subject/object of a higher clause, but there is still a possibility to embed a subject into that clause, should it be considered `ccomp` or `xcomp`?

**Comment by makazhan** [comment]
Here are the analyses of both versions for clarity:
```
1   Мен         I         мен               4    nsubj
2   кино        movie     кино-ACC          3    dobj
3   көргенді    watching  көр-PTCP-ACC      4    xcomp/ccomp(?)
4   ұнатамын    like      ұна-CAUS-AOR-1Sg  0    root
```
```
1   Мен         I         мен                5    nsubj
2   оның        his       ол-GEN             4    nsubj
3   кино        movie     кино-ACC           4    dobj
4   көргенін    watching  көр-PTCP-P3Sg-ACC  5    ccomp
5   ұнатамын    like      ұна-CAUS-AOR-1Sg   0    root
```

**Comment by coltekin** [comment]
The fist one looks like `xcomp`, but second one is certainly not `xcomp` (I have doubts whether we should analyze those as `ccomp`, but this is for another discussion). In Turkish, the subordinating suffixes are different in two cases. In the first case (xcomp) it would be "-mAK", and in the second case it would be "-mA". However, because of morpho-phonological alternations, these suffixes are sometimes indistinguishable on the surface. Are the subordinating suffix in Kazakh the same, or do you also have the duality Turkish has?  If so, that would set a clear guideline.

Here are the parallel examples from Turkish:

```
Ben film   izle-mey-i    severim.
I   movie  watch-mAK-ACC like-AOR-1S. 
I like watching movies.
```

In this case there is no way that anyone else than "I" is interpretted as the subject of the subordinate clause.

```
Ben o-nun       film  izle-me-si-ni       severim.
I   he/she-GEN  movie watch-mA-P3Sg-ACC   like-AOR-1S
I like (that) he/she watches movies.
```

```
Ben senin   film  izle-me-n-i       severim.
I   you-GEN movie watch-mA-P3Sg-ACC like-AOR-1S
I like (that) you watch movies.
```

In the last two examples, even if you leave the pronouns (overt subject) from the subordinate clause out, it is clear that the person doing watching is "he/she" in the first, and "you" in the second example.  So, subordinate clause has a subject that is potentially different than main clause object/subject.

**Comment by makazhan** [comment]
coltekin,
Turkish -mAK corresponds to Kazakh -у /u/: al-mak = ал-у = to take.
About -mA I'm not sure, but we have verb-noun/adj. derivatives (not fully productive) of similar surface form, e.g. көр+ме = see+ме = exhibition; кір+ме = enter+ме = foreign/imported.
In my example I used participle form -ҒАн which corresponds to Turkish -[y]An, but -у (-mAK) could be used as well (arguably even more common). Both forms can be used in both cases (with/without subject).

Anyway, what I meant to clarify is the priority in `xcomp` definition. Is it the fact that:
-- xcomp is a predicative or clausal complement *without its own subject*
or that:
-- the subject of xcomp must be the same as the higher subject or object, with *no other possible interpretation*?

The reason I am asking is that among contributors to Kazakh we have two different interpretations of the definition. For the analysis of the sort of:
```
1   Мен         I         мен               4    nsubj
2   кино        movie     кино-ACC          3    dobj
3   көргенді    watching  көр-PTCP-ACC      4    xcomp/ccomp(?)
4   ұнатамын    like      ұна-CAUS-AOR-1Sg  0    root
```
One possibility is that `watching` doesn't have its own subject, but instead has it referenced by `I`. Hence it must be `xcomp`.
Another reasoning is that `watching` *can* have its own subject, but in the example it simply happens not to have one. We can embed a subject like this:
```
1   Мен         I         мен                5    nsubj
2   оның        his       ол-GEN             4    nsubj
3   кино        movie     кино-ACC           4    dobj
4   көргенін    watching  көр-PTCP-P3Sg-ACC  5    ccomp
5   ұнатамын    like      ұна-CAUS-AOR-1Sg   0    root
```
and it would become apparent that `watching` actually has its own subject (not controlled) so it should be treated as `ccomp`. Something similar to a covert subject case (hence I brought it up earlier).

So, does a *possibility* of embedding a subject into a subject-controlled clausal complement renders it `ccomp`?
For instance, in *I like to drive* `to drive` does not and *cannot* have a subject other than `I`, hence we should have `xcomp(like, drive)`.
However, `driving` also has no internal subject in *I like driving*, but there is a possibility of *I like his driving*. In this case should we use `ccomp(like, driving)` or `xcomp(like, driving)`?

**
Note, I am talking about a *possibility* of embedding, not an actual sentence with such an embedding. To draw a parallel with covert subjects, *Hope you are doing fine* could be *I hope you are doing fine*. Similarly, *I like driving* could be *I like his driving*. I acknowledge the fact that the first two sentences are essentially one thing, but *I like driving* and *I like his driving* are different.

**Comment by coltekin** [comment]
As I understand, the main difference between `xcomp` and `ccomp` is whether  all the arguments of the predicate of the subordinate clause are filled within the clause itself or not. In case of `xcomp`, you need to look at the main/superordinate clause to locate the subject. In `ccomp` the predicate should have its subject argument filled (not necessarily by an overt subject, but, for example, by a suffix on the predicate) without any reference to the higher-level clause.
Then, my question would be whether the P3Sg/P2Sg tags in your examples give away the subject of the subordinate clause even when you do not have an overt subject.
In Turkish examples, even though they are possessive markers on the nominalized phrase/clause, they indicate/agree with the subject of the subordinate clause. As a result, all arguments of the predicate of the subordinate clause are filled within the subordinate clause, and there is no need to look for a missing argument outside. So, if there is a possessive marker after the subordinating morpheme, it is `ccomp`, otherwise it is likely `xcomp` (in case of Turkish, it definitely is `xcomp`, because a predicate has to include a subject marker with or without an overt subject otherwise).

PS. In `ccomp` case, I am inclined towards a different analysis which I will bring up as another discussion point, rather than stealing this thread. I appreciate the comments on parallels with Kazakh and other (Turkic) languages there.

**Comment by makazhan** [comment]
@coltekin,
> As I understand, the main difference between xcomp and ccomp is whether all the arguments of the predicate of the subordinate clause are filled within the clause itself or not. In case of xcomp, you need to look at the main/superordinate clause to locate the subject. In ccomp the predicate should have its subject argument filled (not necessarily by an overt subject, but, for example, by a suffix on the predicate) without any reference to the higher-level clause.

Yes, that's exactly how I understand the definition. 

> Then, my question would be whether the P3Sg/P2Sg tags in your examples give away the subject of the subordinate clause even when you do not have an overt subject.

They certainly do.

> In Turkish examples, even though they are possessive markers on the nominalized phrase/clause, they indicate/agree with the subject of the subordinate clause. As a result, all arguments of the predicate of the subordinate clause are filled within the subordinate clause, and there is no need to look for a missing argument outside.

Same in Kazakh. And when analyzing such cases I would go with `ccomp`, or with `advcl` if a given clause is not a core argument, but definitively not with `xcomp`.
But I am not talking about analyzing *actual sentences* with such clauses. I am talking about sort of a test for `xcomp`.

Suppose we have an English sentence *I love running* (assuming *running* is not a noun).
By definition we should have `xcomp(love, running)`. But! Is it true that `running` does not have its own internal subject? Cannot we say *I love his/your running* while maintaining `running` as a verbal noun whose subject is `his/your`? (May not be grammatical for native speakers, but the point is in embedding a subject). So this "embedding test" shows that `running` *may*, in fact, have an internal subject, even though it *happens* not to have one. Should it then be considered `ccomp` instead of `xcomp`?
In contrast, I *cannot* embed a subject into *I love to run*, so `run` *cannot* have an internal subject and references `I` instead. Hence this is a "cemented" `xcomp` case.

The problem that in Kazakh one can embed a subject in both cases. In fact, we could not find an instance where a clause passed this "embedding test", except cases with secondary predicate (*I consider him smart* → `xcomp(consider, smart)`).
So, if possibility of embedding renders a clause `ccomp`, then it seems that Kazakh does not have `xcomp` in that sense.

P.S. I personally think that clauses before and after embedding constitute two completely different things. Hence the embedding test is not applicable. I just need a confirmation/rejection of my reasoning.

**Comment by jonorthwash** [comment]
One clarification I'd like to make as I'm going through the thread:

Turkish -mAk and -mA both correspond to Kazakh -GAн in this case (and -у in other cases).  It doesn't matter that Kazakh has cognates to -mAk and -mA, since they aren't used the same way.  To answer @coltekin's question regarding this topic, there's no parallel split between verbal noun types in Kazakh.

**Comment by jonorthwash** [comment]
Okay, I understand the results of the discussion up to this point to be that @coltekin and @makazhan agree that `xcomp` is supposed to be a predicative or clausal complement _without its own subject_, whether or not the same structure could be easily modified to have one.  I guess the next step would be for the developers of the UD standard to confirm this.

**Comment by dan-zeman** [comment]
Looks like this thread has been forgotten for a long time, maybe scaring away the unenlightened of us by the “Turkic” label :-)

I tend to agree with @jonorthwash 's summary but I would like to nominate @manning to finally confirm or modify it.

**Comment by msklvsk** [comment]
I’m having trouble interpreting the `xcomp` vs `ccomp` for elided _objects_.

<img width="221" alt="image" src="https://user-images.githubusercontent.com/473012/48676191-f3da5400-eb6b-11e8-914f-616e30bb1063.png">

If the boss instead _ordered_ to start digging, and from the context we know that _The boss ordered **[us]** to start digging_, does this make _to start_ an `xcomp`? (Inspired by #589. Addidng a null `obj` here would definitely allow us to have an `xcomp`.)

**Comment by adam-przepiorkowski** [comment]
That's how we interpreted the guidelines when we developed the
UD_Polish-LFG treebank – there is a brief discussion of a very similar
Polish example in Patejuk and Przepiórkowski 2018
<nlp.ipipan.waw.pl/Bib/pat:prz:18:book.pdf> pp.217–218 (involving Polish
for *ordered*, with a missing controller).  We hope that this is the only
reasonable interpretation of the current guidelines.

Best,
Adam P.

**Comment by amir-zeldes** [comment]
+1 for "ordered to start" = xcomp(ordered, start)!

**Comment by jonorthwash** [comment]
It would be good for @manning to weigh in on this.

**Comment by msklvsk** [comment]
And there is also _the boss demanded **from us** to start_. An `obl` for xsubj?
(I would say yes.)
